File: ruby/lol/lib/lol/guess.rb
def guess_my_number()
  ($big + $small) >> 1
end

def smaller
  $big = guess_my_number
  guess_my_number
end

def bigger
  $small = guess_my_number
  guess_my_number
end

def start_over
  $small = 1
  $big = 100
  guess_my_number
end

File: dotnet/lol/src/guess.cs
using System;

namespace LOL.Guess {
    public class GuessGame
    {
        private int _small = 1;
        private int _big = 100;

        public int GuessMyNumber()
        {
            return (_small + _big) >> 1;
        }

        public int Smaller()
        {
            _big = GuessMyNumber();
            return GuessMyNumber();
        }

        public int Bigger()
        {
            _small = GuessMyNumber();
            return GuessMyNumber();
        }

        public int StartOver()
        {
            _small = 1;
            _big = 100;
            return GuessMyNumber();
        }
    }
}
